bindings/python/gpt4all/gpt4all.py

```diff
@@ -0,0 +1,70 @@
+"""
+Python only API for running all GPT4All models.
+"""
+import os
+
+from . import pyllmodel
+
+
+class GPT4All():
+
+    def __init__(self, model_path: str, model_type: str = None):
+        self.model = None
+
+        # Model type provided for when model is custom
+        if model_type:
+            self.model = GPT4All.get_model_from_type(model_type)
+        # Else get model from gpt4all model filenames
+        else:
+            model_filename = os.path.basename(model_path)
+            self.model = GPT4All.get_model_from_filename(model_filename)
+
+        self.model.load_model(model_path)
+
+    def generate(self, prompt: str, verbose: bool = True):
+        """
+        High level method for running generate on gpt4all models.
+        NOTE: Right now, this is essentially duplicate code but will be useful
+        if underlying models' prompt or generate begin to differ.
+        """
+        return self.model.generate(prompt, verbose=verbose)
+
+    @staticmethod
+    def get_model_from_type(model_type: str) -> pyllmodel.LLModel:
+        # This needs to be updated for each new model
+        # TODO: Might be worth converting model_type to enum
+
+        if model_type == "gptj":
+            return pyllmodel.GPTJModel()
+        elif model_type == "llama":
+            return pyllmodel.LlamaModel()
+        else:
+            raise ValueError(f"No corresponding model for model_type: {model_type}")
+        
+    @staticmethod
+    def get_model_from_filename(model_filename: str) -> pyllmodel.LLModel:
```

Comment:
It would be useful to have an auto-download functionality for our pre-registered models that will places the models in a fixed cache directory (preferably the same directory as the current chat client defaults to)